src/app/todo/components/todoItem/todo-item.component.ts

```diff
@@ -115,12 +157,22 @@ export class TodoItemComponent implements OnInit {
     });
   }
 
-  ngOnInit(): void {
+  async ngOnInit(): Promise<void> {
```

Comment:
you can remove async and promises by utilizing observables for the allUserUsernames thne just pass it to the component in the template with (allUserUsernames$ | async) ?? []